### ISSUE TITLE
Support Rails 7

### ISSUE THREAD

**Issue by thanhtoan1196** [activeadmin/activeadmin#7196]
Can't install activeadmin using rails 7.0.0.alpha2, please help

Logs
```
Fetching gem metadata from https://rubygems.org/.........
Resolving dependencies....
Bundler could not find compatible versions for gem "actionpack":
  In snapshot (Gemfile.lock):
    actionpack (= 7.0.0.alpha2)

  In Gemfile:
    activeadmin was resolved to 0.4.0, which depends on
      meta_search (>= 0.9.2) was resolved to 1.1.0.pre2, which depends on
        actionpack (~> 3.1.0.alpha)

    rails (~> 7.0.0.alpha2) was resolved to 7.0.0.alpha2, which depends on
      actionpack (= 7.0.0.alpha2)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

Bundler could not find compatible versions for gem "rails":
  In snapshot (Gemfile.lock):
    rails (= 7.0.0.alpha2)

  In Gemfile:
    activeadmin was resolved to 1.0.0.pre3, which depends on
      rails (< 5.0, >= 3.2)

    turbo-rails (>= 0.7.11) was resolved to 0.8.3, which depends on
      rails (>= 6.0.0)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.

Bundler could not find compatible versions for gem "railties":
  In snapshot (Gemfile.lock):
    railties (= 7.0.0.alpha2)

  In Gemfile:
    activeadmin was resolved to 1.0.0.pre5, which depends on
      railties (< 5.1, >= 3.2)

    rails (~> 7.0.0.alpha2) was resolved to 7.0.0.alpha2, which depends on
      railties (= 7.0.0.alpha2)

Running `bundle update` will rebuild your snapshot from scratch, using only
the gems in your Gemfile, which may resolve the conflict.
```

**Comment by Fivell** [comment]
related https://github.com/activeadmin/activeadmin/issues/7175

**Comment by tagliala** [comment]
I'm trying to update activeadmin to use rails 7 (https://github.com/tagliala/activeadmin/pull/1), but there are some blocking dependencies:

- [inherited_resources](https://github.com/activeadmin/inherited_resources) activeadmin/inherited_resources#707
- [ransack](https://github.com/activerecord-hackery/ransack) activerecord-hackery/ransack#1266 
- [cucumber-rails](https://github.com/cucumber/cucumber-rails) cucumber/cucumber-rails#523
- [rails-i18n](https://github.com/svenfuchs/rails-i18n) svenfuchs/rails-i18n#957 svenfuchs/rails-i18n#958

Non-blocking, but downgrades to an older version:
- [db-query-matchers](https://github.com/civiccc/db-query-matchers) Looks abandoned :\

**Comment by alejandroperea** [comment]
It is not a blocker but in case we are not able to update `db-query-matchers` I've created a PR to get rid of it: https://github.com/activeadmin/activeadmin/pull/7232

**Comment by ciscolive** [comment]
can team update to rubygem ?

**Comment by tagliala** [comment]
I've updated my comment above by adding [arbre](https://github.com/activeadmin/arbre), that does not allow 7.0 and so it downgrades to 1.2.1

**Comment by baarkerlounger** [comment]
Looks like this branch has issues with the number of args in paginate here https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/views/components/paginated_collection.rb#L111

**Comment by tagliala** [comment]
> Looks like this branch has issues with the number of args in paginate here https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/views/components/paginated_collection.rb#L111

Thanks, will check

**Comment by tagliala** [comment]
For tracking purposes and to prevent new PRs, I opened #7235

**Comment by missy-davies** [comment]
Hi @tagliala, I'd love to help with the effort to make ActiveAdmin compatible with Rails 7. Is there any piece of the puzzle I can take on?

**Comment by tagliala** [comment]
Hi @missy-davies 

> Is there any piece of the puzzle I can take on?

Yes, feel free to take a look at  #7235, there is a "working" branch with failing specs (on CI) and cucumber features

**Comment by missy-davies** [comment]
> Yes, feel free to take a look at #7235, there is a "working" branch with failing specs (on CI) and cucumber features

@tagliala Great, thank you! I'm still working on setting up a virtual machine to be able to clone the repo and run the tests, but once I can see what's failing I'll give it a shot.

**Comment by baarkerlounger** [comment]
> > Looks like this branch has issues with the number of args in paginate here https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/views/components/paginated_collection.rb#L111
> 
> Thanks, will check
> 
> **edit:** I don't know how to replicate (Rails 7.0, Ruby 3.0.3). Is this about pagination in active admin views?
> 
> ![image](https://user-images.githubusercontent.com/556268/146959047-d9f57707-f778-4be4-b585-dab6a0354e7d.png)
> 
> This issue will probably be discovered when it will be possible to run specs on activeadmin

Yep pagination. Looks like a new version of Kaminari has been released now so might be worth bumping that as well.

**Comment by tagliala** [comment]
[Specs are green](https://github.com/tagliala/activeadmin/runs/4637061868?check_suite_focus=true) also with the previous version of Kaminari, if you are exxperiencing a pagination issue, it is probably not covered by specs

**Comment by baarkerlounger** [comment]
Hmm that's odd. Spec coverage claims to cover that 100% and it's a number of arguments error. This is the stack trace:

```
wrong number of arguments (given 2, expected 1)

      # * <tt>:template</tt> - Specify a custom template renderer for rendering the Paginator (receiver by default)
      # * <tt>:ANY_OTHER_VALUES</tt> - Any other hash key & values would be directly passed into each tag as :locals value.
      def paginate(scope, paginator_class: Kaminari::Helpers::Paginator, template: nil, **options)
        options[:total_pages] ||= scope.total_pages
        options.reverse_merge! current_page: scope.current_page, per_page: scope.limit_value, remote: false


kaminari-core (1.2.2) lib/kaminari/helpers/helper_methods.rb:116:in `paginate'
arbre (1.2.1) lib/arbre/element.rb:181:in `method_missing'
activeadmin (12ef798e1a02) lib/active_admin/views/components/paginated_collection.rb:111:in `build_pagination'
activeadmin (12ef798e1a02) lib/active_admin/views/components/paginated_collection.rb:72:in `block in build_pagination_with_formats'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:31:in `block in build_tag'
arbre (1.2.1) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:14:in `div'
activeadmin (12ef798e1a02) lib/active_admin/views/components/paginated_collection.rb:70:in `build_pagination_with_formats'
activeadmin (12ef798e1a02) lib/active_admin/views/components/paginated_collection.rb:54:in `build'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:30:in `block in build_tag'
arbre (1.2.1) lib/arbre/context.rb:92:in `with_current_arbre_element'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:49:in `with_current_arbre_element'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:26:in `build_tag'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:39:in `insert_tag'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:14:in `paginated_collection'
activeadmin (12ef798e1a02) lib/active_admin/views/pages/index.rb:134:in `render_index'
activeadmin (12ef798e1a02) lib/active_admin/views/pages/index.rb:51:in `build_collection'
activeadmin (12ef798e1a02) lib/active_admin/views/pages/index.rb:29:in `block in main_content'
arbre (1.2.1) lib/arbre/element/builder_methods.rb:31:in `block in build_tag' 
```

I also have a repo with specs that fail if that helps reproducing.

**Comment by tagliala** [comment]
> arbre (1.2.1)

I can see a downgraded version of arbre, which is not correct

Please try the following : https://github.com/activeadmin/activeadmin/pull/7235

```
gem 'activeadmin', github: 'tagliala/activeadmin', branch: 'feature/railties-7' # FIXME: revert to stable

gem 'arbre', github: 'activeadmin/arbre' # FIXME: remove
gem 'inherited_resources', github: 'activeadmin/inherited_resources' # FIXME: remove
```

**Comment by baarkerlounger** [comment]
Thanks, you're right, latest Arbre commit fixes it already.

**Comment by muaaz-vf** [comment]
Thank @tagliala adding the above gems does work but whenever we change the code in dev env there is an error afterward,

09:53:56 web.1  | ArgumentError (wrong number of arguments (given 2, expected 5)):
09:53:56 web.1  |   
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/instrumenter.rb:62:in `initialize'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/server_timing.rb:16:in `new'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/server_timing.rb:16:in `block in call'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:225:in `publish'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:80:in `block in publish'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:91:in `block in iterate_guarding_exceptions'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:90:in `each'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:90:in `iterate_guarding_exceptions'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications/fanout.rb:80:in `publish'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/notifications.rb:197:in `publish'
09:53:56 web.1  | activeadmin (12ef798e1a02) lib/active_admin/application.rb:115:in `load!'
09:53:56 web.1  | activeadmin (12ef798e1a02) lib/active_admin/application.rb:228:in `block (2 levels) in attach_reloader'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:445:in `instance_exec'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:445:in `block in make_lambda'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:687:in `block (2 levels) in default_terminator'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:686:in `catch'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:686:in `block in default_terminator'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:200:in `block in halting'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:595:in `block in invoke_before'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:595:in `each'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:595:in `invoke_before'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:106:in `run_callbacks'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/reloader.rb:88:in `prepare!'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/reloader.rb:47:in `block in <class:Reloader>'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:445:in `instance_exec'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:445:in `block in make_lambda'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:271:in `block in simple'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:599:in `block in invoke_after'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:599:in `each'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:599:in `invoke_after'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/callbacks.rb:108:in `run_callbacks'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/execution_wrapper.rb:133:in `run'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/execution_wrapper.rb:129:in `run!'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/reloader.rb:114:in `run!'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/execution_wrapper.rb:74:in `block in run!'
09:53:56 web.1  | <internal:kernel>:90:in `tap'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/execution_wrapper.rb:71:in `run!'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/reloader.rb:63:in `run!'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:12:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/actionable_exceptions.rb:17:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
09:53:56 web.1  | web-console (4.2.0) lib/web_console/middleware.rb:132:in `call_app'
09:53:56 web.1  | web-console (4.2.0) lib/web_console/middleware.rb:28:in `block in call'
09:53:56 web.1  | web-console (4.2.0) lib/web_console/middleware.rb:17:in `catch'
09:53:56 web.1  | web-console (4.2.0) lib/web_console/middleware.rb:17:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/show_exceptions.rb:26:in `call'
09:53:56 web.1  | railties (7.0.0) lib/rails/rack/logger.rb:36:in `call_app'
09:53:56 web.1  | railties (7.0.0) lib/rails/rack/logger.rb:25:in `block in call'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/tagged_logging.rb:99:in `block in tagged'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/tagged_logging.rb:37:in `tagged'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/tagged_logging.rb:99:in `tagged'
09:53:56 web.1  | railties (7.0.0) lib/rails/rack/logger.rb:25:in `call'
09:53:56 web.1  | sprockets-rails (3.4.2) lib/sprockets/rails/quiet_assets.rb:13:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
09:53:56 web.1  | request_store (1.5.0) lib/request_store/middleware.rb:19:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/request_id.rb:26:in `call'
09:53:56 web.1  | rack (2.2.3) lib/rack/method_override.rb:24:in `call'
09:53:56 web.1  | rack (2.2.3) lib/rack/runtime.rb:22:in `call'
09:53:56 web.1  | activesupport (7.0.0) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/server_timing.rb:20:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/executor.rb:14:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/static.rb:23:in `call'
09:53:56 web.1  | rack (2.2.3) lib/rack/sendfile.rb:110:in `call'
09:53:56 web.1  | actionpack (7.0.0) lib/action_dispatch/middleware/host_authorization.rb:137:in `call'
09:53:56 web.1  | railties (7.0.0) lib/rails/engine.rb:530:in `call'
09:53:56 web.1  | puma (5.5.2) lib/puma/configuration.rb:249:in `call'
09:53:56 web.1  | puma (5.5.2) lib/puma/request.rb:77:in `block in handle_request'
09:53:56 web.1  | puma (5.5.2) lib/puma/thread_pool.rb:340:in `with_force_shutdown'
09:53:56 web.1  | puma (5.5.2) lib/puma/request.rb:76:in `handle_request'
09:53:56 web.1  | puma (5.5.2) lib/puma/server.rb:447:in `process_client'
09:53:56 web.1  | puma (5.5.2) lib/puma/thread_pool.rb:147:in `block in spawn_thread'

We have to restart the server and then all seems to work, any workaround will be appreciated. Plus thank you for your great effort.

**Comment by tagliala** [comment]
Any chance to provide a step to step guide to reproduce this issue or a reproducible test case in a public repository?

**Comment by muaaz-vf** [comment]
Sure will share example and steps to produce it today thank you for quick reply.

**Comment by tagliala** [comment]
> Sure will share example and steps to produce it today thank you for quick reply.

No need, reproduced thanks!

**Comment by tagliala** [comment]
@muaaz-vf could you please try to disable `server_timing` ?

```diff
diff --git a/config/environments/development.rb b/config/environments/development.rb
index 62354ff..916e29d 100644
--- a/config/environments/development.rb
+++ b/config/environments/development.rb
@@ -17,7 +17,7 @@ Rails.application.configure do
   config.consider_all_requests_local = true
 
   # Enable server timing
-  config.server_timing = true
+  # config.server_timing = true
 
   # Enable/disable caching. By default caching is disabled.
   # Run rails dev:cache to toggle caching.
```

**Comment by muaaz-vf** [comment]
Thank you much appreciated, anyway I can buy you a cup of coffee :)

**Comment by nquocvuong** [comment]
Hi @tagliala

in my gemfile:
```ruby
gem 'activeadmin', github: 'tagliala/activeadmin', branch: 'feature/railties-7' # FIXME: revert to stable

gem 'arbre', github: 'activeadmin/arbre' # FIXME: remove
gem 'inherited_resources', github: 'activeadmin/inherited_resources' # FIXME: remove
```
and in config/initializers/active_admin.rb
```ruby
...
  config.before_action :abc
...
```
then I got this error:
```text
/lib/active_admin/base_controller/authorization.rb:3:in `<module:ActiveAdmin>': uninitialized constant InheritedResources::Base (NameError)
Did you mean?  Base64
```
if I remove config.before_action, it work

**Comment by tagliala** [comment]
> then I got this error:
> 
> ```
> /lib/active_admin/base_controller/authorization.rb:3:in `<module:ActiveAdmin>': uninitialized constant InheritedResources::Base (NameError)
> Did you mean?  Base64
> ```

Replicated, thanks

**Comment by elliotcm** [comment]
> > then I got this error:
> > ```
> > /lib/active_admin/base_controller/authorization.rb:3:in `<module:ActiveAdmin>': uninitialized constant InheritedResources::Base (NameError)
> > Did you mean?  Base64
> > ```
> 
> Replicated, thanks
> 
> edit: This is related to autoloading. I do not know hot to fix, but I've added a checkbox to known issues at #7235

@tagliala After much digging, I suspect this is due to the issue mentioned in the callout box here:
https://guides.rubyonrails.org/configuring.html#using-initializer-files

My app loads OK after wrapping the contents of `config/initializers/active_admin.rb` in a `Rails.application.config.after_initialize do` block.

**Comment by tagliala** [comment]
Hi @elliotcm, thanks for the feedback

That fix can be addressed in a separate PR like #7262 

I'm not going to create such a PR because I don't know the possible side effects

**Comment by elliotcm** [comment]
> Hi @elliotcm, thanks for the feedback
> 
> That fix can be addressed in a separate PR like #7262
> 
> I'm not going to create such a PR because I don't know the possible side effects

@tagliala Yes I'm not sure there's a fix you can make in the gem(s), but it's a workaround at least for others.  It might even go away when we're not loading the inherited resources gem directly, I'm not savvy enough with the exact timing of Rails loading stuff to be able to suggest a permanent fix.

One reason I wouldn't suggest baking it in to activeadmin is that by moving the activeadmin setup to `after_initialize`, it is set up after controller are loaded (when eager loading in production), so any `before_action`s that you create via the activeadmin config isn't available in the controllers at the time of processing.

In other words, I have this in my AA config:
```
    config.before_action :enforce_2sv
```
and I previously had this in my sessions controller:
```
class Admin::SessionsController < ActiveAdmin::Devise::SessionsController
  skip_before_action :enforce_2sv
end
```
but I've now had to move that to the end of the `after_initialize` block so it happens after AA is set up (so that the action exists to be skipped).
```
Admin::SessionsController.skip_before_action(:enforce_2sv)
```

Quite convoluted but hopefully helps other people get going while things are fixed properly.

**Comment by KevinBongart** [comment]
@tagliala @deivid-rodriguez Since #7235 was merged yesterday (excellent work!), should we consider ActiveAdmin Rails-7 ready and, if so, release a new version?

If not: what's left and where can I find more information about the remaining work? Happy to help.

**Comment by deivid-rodriguez** [comment]
I will release a Rails 7 compatible version next Friday.

I'll close this issue since it was indeed fixed by #7235!

**Comment by jadelekekpmg** [comment]
Thank you so much for resolving this issue. I have been watching it since December.
Appreciate your efforts: @tagliala @deivid-rodriguez

**Comment by mberman84** [comment]
upon first installation in a fresh rails 7 app, I get this error when attempting to navigate to localhost:3000/admin:

`cannot load such file -- sassc`

**Comment by arhaikal** [comment]
Trying to upgrade to rails 7, 
I am running activeadmin 2.11.1 which i think is the latest version but i am still getting the following error: 

`uninitialized constant InheritedResources::Base (NameError)
Did you mean?  Base64`

`gem 'rails', '~> 7.0.0'
ruby '3.0.3'`

anyone is still facing this problem?

**Comment by tagliala** [comment]
> I am running activeadmin 2.11.1 which i think is the latest version but i am still getting the following error:
> 
> `uninitialized constant InheritedResources::Base (NameError) Did you mean? Base64`

This issue should have been fixed by activeadmin/activeadmin#7340

**Comment by arhaikal** [comment]
> > `uninitialized constant InheritedResources::Base (NameError) Did you mean? Base64`
> 
> This issue should have been fixed by #7340

That doesn't seem the case here as the error persists even after removing the before_action from the initializer.
 I am still digging to see what is causing this, so i will make an update if i find something.

**Comment by arhaikal** [comment]
@tagliala I didn't find anything that i can change in my app to mitigate the problem. When the gem is added the error comes from 

`ruby-3.1.1/gems/activeadmin-2.11.1/lib/active_admin/base_controller/authorization.rb:3` which was already mentioned in the issue that you linked

**Comment by heyfadii** [comment]
I'm having the exact same issue as @arhaikal using Version 2.13.0. I'm on ruby 2.7.4 / rails 7.

**Comment by antarr** [comment]
I'm still getting this error when using `ActiveAdmin::BaseController.include(ActiveAdmin::FiltersPersistance)` which I got from [here](https://pawelurbanek.com/active-admin-tips-optimizations).

**Comment by tagliala** [comment]
> I'm still getting this error when using `ActiveAdmin::BaseController.include(ActiveAdmin::FiltersPersistance)` which I got from [here](https://pawelurbanek.com/active-admin-tips-optimizations).

Probably the problem is in 

```rb
ActiveAdmin::BaseController.send(:include, ActiveAdmin::FiltersPersistance)
```

And it is not related to Rails Admin

Please try with 

```rb
ActiveSupport.on_load(:active_admin_controller) do
  ActiveAdmin::BaseController.include ActiveAdmin::FiltersPersistance
end
```

**Comment by antarr** [comment]
@tagliala Thanks, that fixed it

**Comment by ghulamjilani** [comment]
I upgraded my rails version from 6.1 to 7 but facing the following issue related to activeadmin.
I have Implemented many solutions including `Rails.application.config.after_initialize` do 
But, still it is throwing this error..

```
uninitialized constant InheritedResources::Base (NameError)

  class BaseController < ::InheritedResources::Base
                                             ^^^^^^
Did you mean?  Base64
```

**Comment by tagliala** [comment]
@ghulamjilani mai depend on other gems, please check the latest comments in this thread and #7340

> Rails.application.config.after_initialize

The proper solution should involve `ActiveSupport.on_load(:active_admin_controller)` but this may not depend by ActiveAdmin itself
